### PR TITLE
Refactor prop-drilling

### PR DESCRIPTION
**Pull request by AleksandrsKondratjevs**

### CODE REVIEW

File: packages/scandipwa/src/store/Config/Config.action.ts [hunk 1]
```diff
@@ -9,32 +9,16 @@
...
-});
+export const UPDATE_CONFIG_STATE = 'UPDATE_CONFIG_STATE';
 
-/** @namespace Store/Config/Action/updateConfigDevice */
-export const updateConfigDevice = (device: Device): UpdateConfigDeviceAction => ({
-    type: ConfigActionType.UPDATE_CONFIG_DEVICE,
-    device,
+/** @namespace Store/Config/Action/updateConfigState */
+export const updateConfigState = (state: Partial<ConfigStore>): UpdateConfigStateAction => ({
+    type: ConfigActionType.UPDATE_CONFIG_STATE,
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/store/Config/Config.action.ts:22]: Lets stay consist in naming of actions for updating. All others are updateSomething**Store**

File: packages/scandipwa/src/store/Config/Config.type.ts [hunk 2]
```diff
@@ -24,27 +24,15 @@ import { ReviewRatingItem } from 'Query/Review.type';
...
-    device: Device;
-}
-
-export interface UpdateConfigCurrencyAction extends AnyAction {
-    type: ConfigActionType.UPDATE_CURRENT_CURRENCY;
-    selectedCurrency: string;
-}
-
-export type ConfigAction = UpdateConfigAction | UpdateConfigDeviceAction | UpdateConfigCurrencyAction;
+export type ConfigAction = UpdateConfigStateAction;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/store/Config/Config.type.ts:35]: Do we use somewhere this type?

File: packages/scandipwa/src/store/ProductListInfo/ProductListInfo.action.ts [hunk 3]
```diff
@@ -9,31 +9,14 @@
  * @link https://github.com/scandipwa/scandipwa-theme
  */
 
-import { ProductAttributeFilterOptions, ProductsQueryOutput } from 'Query/ProductList.type';
-
 import {
     ProductListInfoActionType,
-    UpdateInfoLoadStatusAction,
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/store/ProductListInfo/ProductListInfo.action.ts:16]: Product list info and product list stores can be unified, please make it.

**Comment by Elena-Evsa** [comment]
Issue 1: Error message 'Your session is over, you are logged out!' shown if user click on 'Log out' in account
image

expected: 
image

**Comment by Elena-Evsa** [comment]
Issue 2: If user is adding to cart 'Baggini Elephant + cross' from wish list error message shown and wishlist' products are not loading https://vimeo.com/784958440/21f8785542

**Comment by Elena-Evsa** [comment]
Issue 3: empty cart shown if user trying to finish order
same for logged in and logged out user
https://vimeo.com/784963179/04fd0cc39a

**Comment by Elena-Evsa** [comment]
issue 4: previous products shown for few seconds when user switching categories or selecting attributes in layered navigation
Previously such issue was too. But now it is much slowlier loading and previous product shown every time
https://vimeo.com/784967201/904c71bf13

**Comment by Elena-Evsa** [comment]
issue 5: placeholders are not shown while products images loading after changing of 'Sort by' type
https://vimeo.com/784967927/cae42136fd

**Comment by Elena-Evsa** [comment]
issue 6: menu absent
image

**Comment by Elena-Evsa** [comment]
issue 7: infinity loader on mobile if user open empty cart

**Comment by Elena-Evsa** [comment]
issue 8: Mobile: placeholders shown after product list widget 
image

**Comment by Elena-Evsa** [comment]
issue 9: Mobile: 'Products are loading...' shown on search page if user enter search query which gives 0 products found

**Comment by Elena-Evsa** [comment]
issue 10: notification absent if user (logged in or not) clicks on 'Notify me when the price drops' or 'Notify me when this product is in stock' 
for example, product https://pr-5305.d25kd40decjpue.amplifyapp.com/default/simple-umbrella.html
https://vimeo.com/785950135/5c23ad833b

**Comment by Elena-Evsa** [comment]
issue 11: Related Products, Up-Sells, and Cross-Sells products absent on PDP and in cart
product for example: https://pr-5305.d25kd40decjpue.amplifyapp.com/default/baggini-y.html
https://vimeo.com/786188847/b83a504bc6

**Comment by AleksandrsKondratjevs** [comment]
Closing due to deprecated code and no activity.

**Pull request closed without merging**